datasets/best/best.py

```diff
@@ -0,0 +1,140 @@
+The test set answers are not provided publicly.
+"""
+
+
+class BestConfig(datasets.BuilderConfig):
+    def __init__(self, **kwargs):
+        """BuilderConfig
+
+        Args:
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(BestConfig, self).__init__(**kwargs)
+
+
+class BestDataset(datasets.GeneratorBasedBuilder):
+
+    _DOWNLOAD_URL = "https://archive.org/download/best_dataset/data.zip"
+    _TRAIN_FOLDER = "train"
+    _TEST_FOLDER = "test"
+
+    _USELESS_TAGS = {"<NE>": "", "</NE>": "", "<AB>": "", "</AB>": ""}
+    # character type mapping from https://github.com/rkcosmos/deepcut/blob/master/deepcut/utils.py
+    _CHAR_TYPES_DICT = {
+        "กขฃคฆงจชซญฎฏฐฑฒณดตถทธนบปพฟภมยรลวศษสฬอ": "c",
+        "ฅฉผฟฌหฮ": "n",
+        "ะาำิีืึุู": "v",  # า ะ ำ ิ ี ึ ื ั ู ุ
+        "เแโใไ": "w",
+        "่้๊๋": "t",  # วรรณยุกต์ ่ ้ ๊ ๋
+        "์ๆฯ.": "s",  # ์  ๆ ฯ .
+        "0123456789๑๒๓๔๕๖๗๘๙": "d",
+        '"': "q",
+        "‘": "q",
+        "’": "q",
+        "'": "q",
+        " ": "p",
+        "abcdefghijklmnopqrstuvwxyz": "s_e",
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZ": "b_e",
+    }
+    _CHAR_TYPE_FLATTEN = {}
+    for ks, v in _CHAR_TYPES_DICT.items():
+        for k in ks:
+            _CHAR_TYPE_FLATTEN[k] = v
+    _CHAR_TYPES = ["b_e", "c", "d", "n", "o", "p", "q", "s", "s_e", "t", "v", "w"]
+
+    BUILDER_CONFIGS = [
+        BestConfig(
+            name="best",
+            version=datasets.Version("1.0.0"),
+            description="",
+        ),
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "fname": datasets.Value("string"),
+                    "char": datasets.Sequence(datasets.Value("string")),
+                    "char_type": datasets.Sequence(datasets.features.ClassLabel(names=self._CHAR_TYPES)),
+                    "is_beginning": datasets.Sequence(datasets.features.ClassLabel(names=["neg", "pos"])),
+                }
+            ),
+            supervised_keys=None,
+            homepage="https://aiforthai.in.th/",
+            citation=_CITATION,
+            license=_LICENSE,
+        )
+
+    def _split_generators(self, dl_manager):
+        arch_path = dl_manager.download_and_extract(self._DOWNLOAD_URL)
+        data_dir = os.path.join(arch_path, "data")
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"filepath": os.path.join(data_dir, self._TRAIN_FOLDER), "split": "train"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={"filepath": os.path.join(data_dir, self._TEST_FOLDER), "split": "train"},
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split):
+        for fname in Path(filepath).rglob("*.txt"):
```

Comment:
```suggestion
        for fname in sorted(Path(filepath).rglob("*.txt")):
```
we need to make the order deterministic

Reply:
Thanks for the review. I think I'll go with `best2009` to conform with the citations. Will reduce size of dummy_data.zip and do sorted file names as well.